### PR TITLE
chore: Deprecation for Item interface in 4.0

### LINKED ISSUE #9511: Drop interface Item
See https://github.com/SAP/spartacus/issues/9113
Drop interface `Item`. Provide update schematics. Add to migration docs.

### PR DESCRIPTION
**Pull request by dydome**
close: #9511

### CODE REVIEW

**Review by marlass — CHANGES_REQUESTED** [review]
Could you also provide migration for that (with removed-public-api-deprecation mechanism)?

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 1]
```diff
@@ -1,39 +1,19 @@
...
-  // {
-  //   node: VARIANT_STYLE_SELECTOR_COMPONENT,
-  //   importPath: SPARTACUS_STOREFRONTLIB,
-  //   comment: `'${VARIANT_STYLE_SELECTOR_COMPONENT}' was moved to @spartacus/product/variants.`,
-  // },
+  // projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts
+  {
+    node: CART_ITEM_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${ITEM}' interface was removed from ${CART_ITEM_COMPONENT}.`,
```
Comment (marlass) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts]: Could you add info about replacement interface? OrderEntry?

**Review by marlass — CHANGES_REQUESTED** [review]

**Review by marlass — APPROVED** [review]

File: projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts [hunk 2]
```diff
@@ -1,39 +1,20 @@
...
-  // {
-  //   node: VARIANT_STYLE_SELECTOR_COMPONENT,
-  //   importPath: SPARTACUS_STOREFRONTLIB,
-  //   comment: `'${VARIANT_STYLE_SELECTOR_COMPONENT}' was moved to @spartacus/product/variants.`,
-  // },
+  // projects/storefrontlib/src/cms-components/cart/cart-shared/cart-item/cart-item.component.ts
+  {
+    node: CART_ITEM_COMPONENT,
+    importPath: SPARTACUS_STOREFRONTLIB,
+    comment: `'${ITEM}' interface was removed from ${CART_ITEM_COMPONENT}. User ${ORDER_ENTRY} instad.`,
```
Comment (Platonn) [inline projects/schematics/src/migrations/4_0/removed-public-api-deprecations/removed-public-api-deprecation.ts:16]:
```suggestion
    comment: `'${ITEM}' interface was removed from ${CART_ITEM_COMPONENT}. Use ${ORDER_ENTRY} instad.`,
```

**Comment by marlass** [comment]
Merged to `epic/4-0-deprecations`

**Pull request closed without merging**